tsconfig.json

```diff
@@ -0,0 +1,44 @@
+{
+  "compilerOptions": {
+    "target": "es2018",
```

Comment:
I don't really know anything about this file, but some naive questions:

es2018 - this works in browsers?
Any thing I specifically need to know about here?

Reply:
I think we could use some lower target, but this is the default when converting the project to typescript and the code seems to be working. With a higher target we will have a transpiled JS code closest to typescript without crazy workarounds.

Reply:
You could go as high as `es2019` given that beekeeper uses chromium 80 (through electron 8.2.5) and a minimum node version of 12. Unless you wanted to support a lower version of node or electron, there is no reason to not make this as high as possible, using a combination of https://kangax.github.io/compat-table/es2016plus/ and https://node.green/ to see what features the environment matches.

Reply:
I would stay this value for now, based that this is the default. But after this being merged we can play with this value for tests and check if we have any gain with higher/lower values.